lib/workers/branch/index.ts

```diff
@@ -320,6 +325,74 @@ export async function processBranch(
     } else {
       logger.debug('No updated lock files in branch');
     }
+
+    if (global.trustLevel === 'high') {
+      logger.debug('Checking for post-upgrade hooks');
+      if (
+        is.nonEmptyArray(config.postUpgradeTasks) &&
+        is.nonEmptyArray(global.allowedPostUpgradeTasks)
```

Comment:
```suggestion
        is.nonEmptyArray(config.allowedPostUpgradeTasks)
```

If `admin:true` this is not overridable by repo config.